src/t8_geometry/t8_geometry_helpers.c

```diff
@@ -244,3 +266,156 @@ t8_geom_get_edge_vertices (const t8_eclass_t tree_class,
+    if (ref_coords[0] == ref_opposite_vertex[0]) {
+      ref_intersection[0] = 1;
+      ref_intersection[1] = 1;
+      break;
+    }
+    /* If the reference point lies somewhere on edge 2, the intersection has to be at (0,0). */
+    else if (ref_coords[1] == ref_opposite_vertex[1]) {
+      ref_intersection[0] = 0;
+      ref_intersection[1] = 0;
+      break;
+    }
+    else {
+      /* intersectionX = (x1y2-y1x2)(x3-x4)-(x1-x2)(x3y4-y3x4)
+       *                 /(x1-x2)(y3-y4)-(y1-y2)(x3-x4)
+       * intersectionY = (x1y2-y1x2)(y3-y4)-(y1-y2)(x3y4-y3x4)
+       *                 /(x1-x2)(y3-y4)-(y1-y2)(x3-x4)
+       * 
+       * x1=0 y1=0 x2=1 y2=1 x3=ref_coords[0] y3=ref_coords[1] x4=ref_opposite_vertex[0] y4=ref_opposite_vertex[1]
+       * 
+       * Since the intersection point lies on edge 2, which has a slope of 1, the x and the y value has to be equal
+       */
+      ref_intersection[0] = ref_intersection[1] =
+        ((ref_coords[0] * ref_opposite_vertex[1] -
+          ref_coords[1] * ref_opposite_vertex[0])
+         / -(ref_coords[1] - ref_opposite_vertex[1]) +
+         (ref_coords[0] - ref_opposite_vertex[0]));
+      break;
+    }
+  case 2:                      /* edge 2 */
+    /* If the reference point lies somewhere on edge 0, the intersection has to be at (1,0). */
+    if (ref_coords[0] == ref_opposite_vertex[0]) {
+      ref_intersection[0] = 1;
+      ref_intersection[1] = 0;
+      break;
+    }
+    /* If the reference point is equal to the opposite vertex, the intersection has to be at (0,1). */
+    else if (ref_coords[1] == ref_opposite_vertex[1]) {
+      ref_intersection[0] = 0;
+      ref_intersection[1] = 1;
+    }
+    else {
+      /* intersectionX = (x1y2-y1x2)(x3-x4)-(x1-x2)(x3y4-y3x4)
+       *                 /(x1-x2)(y3-y4)-(y1-y2)(x3-x4)
+       * intersectionY = (x1y2-y1x2)(y3-y4)-(y1-y2)(x3y4-y3x4)
+       *                 /(x1-x2)(y3-y4)-(y1-y2)(x3-x4)
+       * 
+       * x1=0 y1=0 x2=1 y2=0 x3=ref_coords[0] y3=ref_coords[1] x4=ref_opposite_vertex[0] y4=ref_opposite_vertex[1]
+       * 
+       * Since the intersection point lies on edge 2, which has a slope of 1, the x and the y value has to be equal
+       */
+      ref_intersection[0] =
+        (ref_coords[0] * ref_opposite_vertex[1] -
+         ref_coords[1] * ref_opposite_vertex[0])
+        / (-(ref_coords[1] - ref_opposite_vertex[1]));
+      /* Since edge 2 is horizontal, the y value of the intersection always has to be 0. */
+      ref_intersection[1] = 0;
+      break;
+    }
+  default:
+    SC_ABORT_NOT_REACHED ();
+    break;
+  }
+}
+
+void
+t8_geom_get_triangle_scaling_factor (int edge_index,
+                                     const double *tree_vertices,
+                                     const double *glob_intersection,
+                                     const double *glob_ref_point,
+                                     double *scaling_factor)
```

Comment:
Since the scaling factor isn't an array, you can just use a normal return like `double
t8_geom_get_triangle_scaling_factor `